woocommerce/wc-admin
# Add import endpoint and controller

Fixes (part of) #1850

* Adds the new import endpoint
* Allows regenerating data by number of days and skipping existing records
* Hooks up the settings page regenerate call to the new endpoint

### Thoughts / Questions

* ~This PR removes `customer_lookup_batch_init()` and relevant dependent actions.  It seems like this is safe since customers are now processed during orders, but want to double check.  This also needs to be removed to see benefit of the speed improvements by only processing days or unprocesse orders.  /cc @jeffstieler~
* Should the regenerate query with days/skip options be placed in options for retrieval instead of passed down through multiple functions?  I'm thinking passing as query_args is safe enough, but open to suggestions here.

### Screenshots
<img width="819" alt="Screen Shot 2019-04-11 at 11 17 13 AM" src="https://user-images.githubusercontent.com/10561050/55952950-69ff6480-5c8d-11e9-9e41-c9b77f56d487.png">

### Detailed test instructions:

1. Make a POST request to `wp-json/wc/v4/reports/import`.
2. Check that your data regeneration has scheduled actions.
3. Pass the following params to the API and test that it limits the number of orders processed.
  - `days` (int) - Number of days to process prior to today.
  - `skip_existing` (bool) - Skips existing records if already imported.
4. Make sure all tests pass.
5.  Make sure customer data is still updated correctly on order import.
6. Make sure the import option at `wp-admin/admin.php?page=wc-admin#/analytics/settings` still works as expected.
